### PR TITLE
[Mellanox] Changed platform summary to reflect HW revision from vpd_data

### PR DESCRIPTION
**Pull request by skr31**
#### Why I did it
The command 'show platform summary' reflected the wrong HW revision.
In order to have the correct HW revision, we need to read it from vpd_data file that HW management provide.
##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
Changed the source from which the HW revision is taken
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Comment by dprital** [comment]
Please elaborate about the description.

For example:

The command 'show platform summary' reflected the wrong HW revision. In order to have the correct HW revision, we need to read it from vpd_data file that .......

File: platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py [hunk 1]
```diff
@@ -41,22 +41,8 @@
...
-DMI_LOC = "Location In Chassis"
-DMI_TABLE_MAP = {
-                    DMI_PRODUCT_NAME: 0,
-                    DMI_MANUFACTURER: 1,
-                    DMI_VERSION: 2,
-                    DMI_SERIAL: 3,
-                    DMI_ASSET_TAG: 4,
-                    DMI_LOC: 5
-                }
+VPD_DATA_FILE = "/var/run/hw-management/eeprom/vpd_data"
```
Comment (dprital) [inline platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py:44]: Need to make sure that the file is available for the current version of HW MNG and above. Please check.

File: platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py [hunk 2]
```diff
@@ -933,12 +919,12 @@ def get_revision(self):
...
+        if self.vpd_data is None:
+            self.vpd_data = self._parse_vpd_data(VPD_DATA_FILE)
 
-        return self.dmi_data.get(DMI_VERSION, "N/A")
+        return self.vpd_data.get(REVISION, "N/A")
 
-    def _parse_dmi(self, filename):
+    def _parse_vpd_data(self, filename):
         """
         Read DMI data chassis data and returns a dictionary of values
```
Comment (dprital) [inline platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py]: Please update comment. Read VPD.....

File: platform/mellanox/mlnx-platform-api/tests/test_chassis.py [hunk 3]
```diff
@@ -324,7 +324,7 @@ def test_module(self):
         assert len(module_list) == 3
         assert chassis.module_initialized_count == 3
 
-    def test_revision_permission(self):
+    """def test_revision_permission(self):
```
Comment (dprital) [inline platform/mellanox/mlnx-platform-api/tests/test_chassis.py]: To be removed ?

**Review by dprital — CHANGES_REQUESTED** [review]

File: platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py [hunk 4]
```diff
@@ -950,15 +936,14 @@ def _parse_dmi(self, filename):
...
 
-            with open(filename, "rb") as fileobj:
-                data = fileobj.read()
-
-            body = data[DMI_HEADER_LEN:]
-            records = body.split(b'\x00')
-
-            for k, v in DMI_TABLE_MAP.items():
-                result[k] = records[v].decode("utf-8")
+            with open(filename, "r") as vpd_file:
```
Comment (Junchao-Mellanox) [inline platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py]: please use utils.read_key_value_file

Comment (skr31) [reply platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py]: Fixed

File: platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py [hunk 5]
```diff
@@ -933,14 +919,14 @@ def get_revision(self):
         Returns:
             string: Revision value of device
         """
-        if self.dmi_data is None:
-            self.dmi_data = self._parse_dmi(DMI_FILE)
+        if self.vpd_data is None:
```
Comment (Junchao-Mellanox) [inline platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py]: if not self.vpd_data. This will check both None and empty dictionary.

Comment (skr31) [reply platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py]: Fixed

**Review by Junchao-Mellanox — APPROVED** [review]

**Pull request closed without merging**